nic_test.go

```diff
@@ -31,6 +32,11 @@ func assertCanCreateNIC(
 	if nic.VMID() != vm.ID() {
 		t.Fatalf("VM ID mismatch between NIC and VM (%s != %s)", nic.VMID(), vm.ID())
 	}
+	if params != nil {
+		if nic.Mac() != params.Mac() {
+			t.Fatalf("Failed to create NIC with custom mac address: %s", nic.Mac())
```

Comment:
```suggestion
		if params.Mac() != "" && params.Mac() != nic.Mac() {
			t.Fatalf("Failed to create NIC with custom mac address. Expected '%s', but created mac is '%s'", params.Mac(), nic.Mac())
```
The additional empty string check on the mac param is required here. If the mac param is not set, a mac for the nic will be generated anyway when testing against a real oVirt engine - and cause this assert func to fail (for example in [TestVMNICCreation](https://github.com/oVirt/go-ovirt-client/blob/main/nic_create_test.go#L21-L26)